src/erasure-code/jerasure/ErasureCodeJerasure.cc

```diff
@@ -191,8 +191,6 @@ int ErasureCodeJerasureReedSolomonVandermonde::parse(ErasureCodeProfile &profile
   if (w != 8 && w != 16 && w != 32) {
     *ss << "ReedSolomonVandermonde: w=" << w
 	<< " must be one of {8, 16, 32} : revert to " << DEFAULT_W << std::endl;
-    profile["w"] = "8";
```

Comment:
the commit message is misleading. it sounds like it's a refactor but it changes the behaviors of multiple erasure plugins without mentioning what is being changed.

before this change, the parameter of some plugins is corrected using the default setting and `-EINVAL` is returned. in the case of `ErasureCodeJerasureLiber8tion`, `m` and `w` are simply overwritten by the default values.

after this change, the parameter of those plugins are not corrected anymore. and `ErasureCodeJerasureLiber8tion` 's parameters are not fixed or overwritten .

Reply:
> after this change, the parameter of those plugins are not corrected anymore. and ErasureCodeJerasureLiber8tion 's parameters are not fixed or overwritten .

I realized that it changes the behaviors, however, `-EINVAL` is returned in this case, and the `profile` will not be created if we use command `ceph osd erasure-code-profile set xxx`, so it sounds meaningless whether `m` and `w` is corrected or not.  right?

Reply:
i think you missed my comment.

> after this change, the parameter of those plugins are not corrected anymore. and `ErasureCodeJerasureLiber8tion` 's parameters are not fixed or overwritten .

Reply:
ok,,, then I dropped the second commit, and changed the first one, could you re-check ?